lua-libraries/lunamark/reader/markdown.lua

```diff
@@ -1661,6 +1665,33 @@ function M.new(writer, options)
                   * larsers.table_caption^-1
                   / writer.table
 
+  ------------------------------------------------------------------------------
+  -- Math
+  ------------------------------------------------------------------------------
+
+  -- Closing $ must have a non-space character immediately to its left,
+  --       and must not be followed immediately by a digit.
+  -- NOTE: For display math, "there can be no blank lines between the opening and closing $$ delimiters",
+  -- Does this implies blank lines are acceptable in inline math? For now, assume it's not the case,
+  -- and not even a line break (which would be allowed in $$ per the above)
+  larsers.inlinemathtail = #(parsers.linechar -parsers.space) * parsers.dollar * #(-parsers.digit)
+  larsers.inlinemath = parsers.between(C((parsers.linechar - parsers.dollar)^1),
+                                       -- Opening $ mush have non space character immediately to its right
+                                       parsers.dollar * #(parsers.any -#parsers.space),
+                                       larsers.inlinemathtail)
+
+  -- The $$ delimiters may be separated from the formula by whitespace.
+  -- NOTE: It's not clear from the description whether the following digit restriction still applies,
+  -- but doesn't seem to be the case in Pandoc's source (mathDisplayWith in Text/Pandoc/Parsing/Math.hs)
+  larsers.displaymathdelim = parsers.dollar * parsers.dollar
+  larsers.displaymath     = larsers.displaymathdelim
+                            -- There can be no blank lines between the opening and closing $$ delimiters.
+                          * C((parsers.any -parsers.blankline^2 -parsers.dollar -larsers.displaymathdelim)^1)
```

Comment:
Github and Pandoc accept a single `$` in display math: `$$ e = $x $$` is $$ e = $x $$
So my `-parsers.dollar` is probably wrong here.

(And some say Markdown is simple. Folks you got mad at one point ;)

Reply:
Scrap that. Both Pandoc and Github d**o not** seem to accept a single $ in display math.